runtime/lua/vim/filetype.lua

```diff
@@ -1007,6 +1036,156 @@ local extension = {
       return 'text'
     end
   end,
+  cmd = function(path, bufnr)
+    if getline(bufnr, 1):find('^/%*') then
+      return 'rexx'
+    end
+    return 'dosbatch'
+  end,
+  rul = function(path, bufnr)
+    require('vim.filetype.detect').rul(bufnr)
+  end,
+  cpy = function(path, bufnr)
+    if getline(bufnr, 1):find('^##') then
+      return 'python'
+    end
+    return 'cobol'
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find('^%s*<!') then
+      return 'dsl'
+    end
+    return 'structurizr'
+  end,
+  edf = 'edif',
+  edfi = 'edif',
+  edo = 'edif',
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find('^%s*%(%s*edif>') then
```

Comment:
`\>` matches the end of a word in Vim, so should use `matchregex` function:
```lua
if matchregex(line, [[\c^\s*(\s*edif\>]])
```

Reply:
We don't have `matchregex` function in `filetype.lua`, so should I move this validation to `detect.lua` ?

Reply:
I think the whole check can be moved to `detect.lua`, do you agree @gpanders?

Reply:
Yes, good idea to avoid duplicating these helpers.